File: 30daysOFcode_TypeScript/day 08.ts
function processData(input: string) {
    let inputArr: Array<string> = input.split('\n');

    let phoneBook = {};
    
    for (let i: number = 1; i <= parseInt(inputArr[0]); i++) {
        let contact: Array<string> = inputArr[i].split(' ');
        phoneBook[contact[0]] = parseInt(contact[1]);
    }

    for (let i: number = parseInt(inputArr[0]) + 1; i < inputArr.length; i++) {
        if (phoneBook[inputArr[i]]) {
            console.log(inputArr[i] + '=' + phoneBook[inputArr[i]]);
        }
        else {
            console.log('Not found');
        }
    }

}

File: 30daysOFcode_Swift/day 08.swift
// Day 08 swift 
// read input
var n = Int(readLine()!)!
// phone number
var phoneBook = [String: String]()

for i in 0...n - 1 {
    // read contacts into phone book
    var contact = readLine()!.characters.split(separator: " ").map(String.init)
    phoneBook[contact[0]] = contact[1]
}

var queryName = readLine()
while queryName != nil {
    if let number = phoneBook[queryName!] {
        print("\(queryName!)=\(phoneBook[queryName!]!)")
    }
    else {
        print("Not found")
    }
    queryName = readLine()
}
